src/urllib3/__init__.py

```diff
@@ -102,4 +106,14 @@ def request(
     The method does not accept low-level ``**urlopen_kw`` keyword arguments.
     """
```

Comment:
I wonder if we're getting to the point where we should more explicitly (and repeatedly) document all of our parameters here?

It's a pain to have to click through N layers of "... and the rest of the args come from X" (top-level request -> PoolManager request -> ConnectionPool urlopen -> HTTPResponse from_httplib).

I feel like the top-level request helper will be a great entry-point for most people and learning about what the various parameters do off the bat would help beginners a lot.

(Also don't think this is blocking of this PR, but would be nice)

Reply:
Yes, that'd be a great idea. Btw apart from the added args, I found a few more args like `decode_content` , `enforce_content_length` etc. But I am not sure if they have any importance or not.

Reply:
Yea let's not be super exhaustive about adding all the extra args, we can be incremental about that. :)